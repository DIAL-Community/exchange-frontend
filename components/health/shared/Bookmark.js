import { useCallback, useContext } from 'react'
import { useRouter } from 'next/router'
import { useIntl } from 'react-intl'
import { useMutation } from '@apollo/client'
import { FaRegBookmark } from 'react-icons/fa6'
import { useUser } from '../../../lib/hooks'
import { ToastContext } from '../../../lib/ToastContext'
import { ADD_BOOKMARK } from '../../../components/shared/mutation/bookmark'

const Bookmark = ({ object, sharableLink, objectType }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const { user } = useUser()
  const { locale, pathname } = useRouter()
  const { showSuccessMessage, showFailureMessage } = useContext(ToastContext)

  const [addBookmark, { reset }] = useMutation(ADD_BOOKMARK, {
    onCompleted: (data) => {
      const { addBookmark: response } = data
      if (response?.bookmark && response?.errors?.length === 0) {
        showSuccessMessage(format('toast.addBookmark.success'))
      } else {
        showFailureMessage(format('toast.addBookmark.failure'))
        reset()
      }
    },
    onError: () => {
      showFailureMessage(format('toast.addBookmark.failure'))
      reset()
    }
  })

  const bookmarkThis = () => {
    if (user && objectType) {
      const { userEmail, userToken } = user
      addBookmark({
        variables: {
          data: object?.id
            ? object.id
            : sharableLink
              ? sharableLink()
              : pathname,
          type: objectType
        },
        context: {
          headers: {
            'Accept-Language': locale,
            Authorization: `${userEmail} ${userToken}`
          }
        }
      })
    }
  }

  return (
    <div className='flex flex-col gap-3 py-3'>
      <div className='text-black font-semibold'>
        {format('ui.bookmark.title')}
      </div>
      <button type='button' onClick={bookmarkThis} className='group flex flex-row gap-x-3'>
        <FaRegBookmark
          className='w-10 h-10 bg-health-red fill-white rounded-full rotate-45'
          viewBox="-200 -200 800 900"
        />
        <div className='my-auto border-b border-transparent group-hover:border-dial-slate-500'>
          <div className='text-sm text-dial-stratos text-left'>
            {format('ui.bookmark.bookmarkThis')}
          </div>
        </div>
      </button>
      <div className='text-sm text-dial-stratos'>
        {format('ui.bookmark.subtitle')}
      </div>
    </div>
  )
}

export default Bookmark
