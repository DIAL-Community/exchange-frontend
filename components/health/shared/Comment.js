import { useCallback } from 'react'
import { useIntl } from 'react-intl'

const Comment = ({ entityKey, scrollRef }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const scrollToComment = () => {
    if (scrollRef && scrollRef.current) {
      const scrollTargetRef = scrollRef.current.find(ref => ref.value === 'ui.comment.label')
      scrollTargetRef?.ref.current.scrollIntoView({
        behavior: 'smooth',
        block: 'center',
        inline: 'start'
      })
    }
  }

  return (
    <div className='flex flex-col gap-3 py-3'>
      <div className='text-black font-semibold'>
        {format('ui.comment.label')}
      </div>
      <div className='text-sm text-black'>
        {format('ui.comment.description', { entity: format(entityKey) })}
      </div>
      <div className='flex text-white'>
        <div className='bg-health-blue rounded-md text-sm'>
          <button type='button' className='px-5 py-3' onClick={scrollToComment}>
            {format('ui.comment.buttonTitle')}
          </button>
        </div>
      </div>
    </div>
  )
}

export default Comment
