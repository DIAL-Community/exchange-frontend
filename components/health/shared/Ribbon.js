import { useCallback } from 'react'
import { useIntl } from 'react-intl'
import Breadcrumb from './Breadcrumb'

const Ribbon = ({ ribbonBg, titleKey, titleImage, titleColor, breadcrumb }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  return (
    <div className={`${ribbonBg} ribbon-outer rounded-b-[32px] z-40`}>
      <div className='flex flex-col items-center gap-y-1'>
        <div className='mr-auto px-4 lg:px-8 xl:px-56 my-3'>
          <Breadcrumb slugNameMapping={breadcrumb} />
        </div>
        <div className='ribbon-inner w-full my-auto'>
          <div className='flex px-4 lg:px-8 xl:px-56'>
            <div className='basis-full lg:basis-3/4 flex flex-col gap-4'>
              <div className='flex gap-4 my-auto'>
                {titleImage}
                <div className={`text-2xl font-base ${titleColor} my-auto flex-grow`}>
                  {format(titleKey)}
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default Ribbon
