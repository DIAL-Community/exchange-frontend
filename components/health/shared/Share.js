import { useCallback } from 'react'
import { useIntl } from 'react-intl'
import { AiFillTwitterCircle } from 'react-icons/ai'
import { TiSocialLinkedin } from 'react-icons/ti'
import { CiMail } from 'react-icons/ci'
import { FaWhatsapp } from 'react-icons/fa'

const Share = () => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  return (
    <div className='flex flex-col gap-3 py-3'>
      <div className='text-black font-semibold'>
        {format('ui.share.title')}
      </div>
      <div className='flex flex-row gap-3 align-center'>
        <AiFillTwitterCircle
          className='w-11 h-11 fill-health-blue'
        />
        <TiSocialLinkedin
          className='w-10 h-10 bg-health-blue fill-white rounded-full'
          viewBox="-3 -3 30 30"
        />
        <FaWhatsapp
          className='w-10 h-10 bg-health-blue fill-white rounded-full'
          viewBox="-80 -80 600 680"
        />
        <CiMail
          className='w-10 h-10 bg-health-blue fill-white rounded-full'
          viewBox="-6 -6 36 36"
        />
      </div>
    </div>
  )
}

export default Share
