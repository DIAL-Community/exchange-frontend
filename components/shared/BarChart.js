import { BarElement, CategoryScale, Chart, LinearScale } from 'chart.js'
import { Bar } from 'react-chartjs-2'

Chart.register(
  CategoryScale,
  LinearScale,
  BarElement
)

const BarChart = ({ labels, values, maxScaleValue, horizontal = false, fontSize = 16 }) => {
  const chartData = {
    labels,
    datasets: [
      {
        data: values,
        backgroundColor: 'rgba(63, 158, 221, 0.2)',
        borderColor: 'rgba(63, 158, 221, 1)',
        borderWidth: 1
      }
    ]
  }

  const chartOptions = {
    maintainAspectRatio: false,
    scale: {
      max: maxScaleValue,
      font: {
        size: fontSize
      },
      color: '#46465a'
    },
    indexAxis: horizontal && 'y'
  }

  return <Bar data={chartData} options={chartOptions} />
}

export default BarChart
