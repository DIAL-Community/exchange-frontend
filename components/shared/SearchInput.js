import { HiOutlineSearch } from 'react-icons/hi'
import { MdClose } from 'react-icons/md'
import classNames from 'classnames'

const EMPTY_VALUE = ''

export const SearchInput = ({
  value,
  placeholder,
  onChange,
  onBlur,
  onSearchIconClick = null,
  className,
  ...otherProps
}) => (
  <div className='flex items-center'>
    <input
      {...otherProps}
      type='text'
      value={value}
      onChange={onChange}
      onBlur={onBlur}
      placeholder={placeholder}
      className={classNames(className, { 'pr-20': onSearchIconClick, 'pr-10': !onSearchIconClick }, 'w-full')}
      data-testid='search-input'
    />
    <div className={classNames({ '-ml-[4.5rem]': onSearchIconClick, '-ml-8': !onSearchIconClick }, 'flex mr-2')}>
      <span className='my-auto pr-1 bg-white text-2xl text-dial-gray-dark border-dial-gray-dark'>
        <MdClose
          data-testid='clear-icon-button'
          onClick={() => onChange({ target: { value: EMPTY_VALUE } })}
          className='cursor-pointer'
        />
      </span>
      {onSearchIconClick && (
        <span className='my-auto pl-1 border-l bg-white text-2xl text-dial-gray-dark border-dial-gray-dark'>
          <HiOutlineSearch
            data-testid='search-icon-button'
            onClick={onSearchIconClick}
            className='cursor-pointer'
          />
        </span>
      )}
    </div>
  </div>
)
