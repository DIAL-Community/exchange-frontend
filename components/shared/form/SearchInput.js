import { MdClose } from 'react-icons/md'

export const SearchInput = ({ value, placeholder, onChange }) => (
  <label htmlFor='search-bar' className='relative'>
    <input
      id='search-bar'
      type='text'
      className='w-full border text-sm'
      placeholder={placeholder}
      onChange={onChange}
      value={value}
    />
    <MdClose
      onClick={() => onChange({ target: { value: '' } })}
      className='absolute top-1/2 transform -translate-y-1/2 right-2'
    />
  </label>
)
