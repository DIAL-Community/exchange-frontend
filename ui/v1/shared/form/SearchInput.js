import { MdClose } from 'react-icons/md'

export const SearchInput = ({ value, placeholder, onChange }) => (
  <label htmlFor='search-bar' className='relative'>
    <input
      id='search-bar'
      type='text'
      className='form-input w-full border'
      placeholder={placeholder}
      onChange={onChange}
      value={value}
    />
    <MdClose
      onClick={() => onChange({ target: { value: '' } })}
      className='absolute top-1/2 transform -translate-y-1/2 right-2'
    />
  </label>
)
