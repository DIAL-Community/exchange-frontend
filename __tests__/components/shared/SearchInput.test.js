import { render, fireEvent } from '@testing-library/react'
import { SearchInput } from '../../../components/shared/SearchInput'

describe('Unit test for the Search Input component.', () => {
  const TEST_ID = 'search-input'

  test('Should match snapshot - without search icon.', () => {
    const onChange = jest.fn()
    const { container } = render(<SearchInput value='test' onChange={onChange} />)
    expect(container).toMatchSnapshot()
  })

  test('Should match snapshot - with search icon.', () => {
    const onChange = jest.fn()
    const onSearchIconClick = jest.fn()
    const { container } = render(<SearchInput value='test' onChange={onChange} onSearchIconClick={onSearchIconClick} />)
    expect(container).toMatchSnapshot()
  })

  test('Should call onChange function - updated value.', () => {
    const onChange = jest.fn()
    const input = render(<SearchInput value='test' onChange={onChange} />).getByTestId(TEST_ID)
    fireEvent.change(input, { target: { value: 'updated' }})
    expect(onChange).toBeCalledTimes(1)
  })

  test('Should call onChange function - clear.', () => {
    const onChange = jest.fn()
    const clearIconButton = render(<SearchInput value='test' onChange={onChange} />).getByTestId('clear-icon-button')
    fireEvent.click(clearIconButton)
    expect(onChange).toBeCalledTimes(1)
  })

  test('Should call onSearchIconClick function.', () => {
    const onChange = jest.fn()
    const onSearchIconClick = jest.fn()
    const searchIconButton = render(<SearchInput value='test' onChange={onChange} onSearchIconClick={onSearchIconClick} />).getByTestId('search-icon-button')
    fireEvent.click(searchIconButton)
    expect(onSearchIconClick).toBeCalledTimes(1)
  })
})
