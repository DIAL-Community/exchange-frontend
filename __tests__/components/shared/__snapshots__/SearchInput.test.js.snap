// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for the SearchInput component. Should match snapshot - with search icon. 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <input
      class="pr-20 w-full"
      data-testid="search-input"
      type="text"
      value="test"
    />
    <div
      class="-ml-[4.5rem] flex mr-2"
    >
      <span
        class="my-auto pr-1 bg-white text-2xl text-dial-gray-dark border-dial-gray-dark"
      >
        <svg
          class="cursor-pointer"
          data-testid="clear-icon-button"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </span>
      <span
        class="my-auto pl-1 border-l bg-white text-3xl text-dial-gray-dark border-dial-gray-dark"
      >
        <svg
          class="cursor-pointer"
          data-testid="search-icon-button"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`Unit test for the SearchInput component. Should match snapshot - without search icon. 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <input
      class="pr-10 w-full"
      data-testid="search-input"
      type="text"
      value="test"
    />
    <div
      class="-ml-8 flex mr-2"
    >
      <span
        class="my-auto pr-1 bg-white text-2xl text-dial-gray-dark border-dial-gray-dark"
      >
        <svg
          class="cursor-pointer"
          data-testid="clear-icon-button"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
